RevenueCatUI/View+PresentPaywall.swift

```diff
@@ -144,6 +154,13 @@ private struct PresentingPaywallModifier: ViewModifier {
 
                         self.close()
                     }
+                    .onPurchaseCompleted { customerInfo in
+                        self.restoreCompleted?(customerInfo)
+
+                        if !self.shouldDisplay(customerInfo) {
+                            self.close()
+                        }
```

Comment:
Data driven UI FTW: we can make it dismiss automatically as well if the condition becomes `false` after restoring 🎉